src/sql-parser/src/parser.rs

```diff
@@ -1859,6 +1861,31 @@ impl<'a> Parser<'a> {
         Ok(compression)
     }
 
+    fn parse_create_connector(&mut self) -> Result<Statement<Raw>, ParserError> {
+        self.expect_keyword(CONNECTOR)?;
+        let if_not_exists = self.parse_if_not_exists()?;
+        let name = self.parse_object_name()?;
+        self.expect_keyword(FOR)?;
+        let connector = match self.expect_one_of_keywords(&[KAFKA])? {
+            Keyword::Kafka => {
+                self.expect_keyword(BROKER)?;
+                let broker = self.parse_literal_string()?;
+                self.expect_keyword(WITH)?;
```

Comment:
I don't understand requiring `WITH` to be present, but allowing its options to be empty. It seems like you want to make it optional, and set `with_options` to some empty default value. What do you think about that? Am I missing something?

Reply:
I believe that `WITH` should be required and have at least one specified option since otherwise it makes no sense to use a connector. This also happens to simplify the implementation but I think its the right choice based on the actual goal of using a `CONNECTOR` instead of just specifying the broker inline.

Ultimately I think this entire syntax should change and the plan is to use the fact that connectors are still experimental as a chance to get away from using with options and instead specify in the grammar which things are required and which are optional. Until then though I think its appropriate for `CONNECTORS` to only support the intended use case where a set of options beyond just the broker URI will be re-used.

Reply:
Ah, this was my mistake--I didn't realize `parse_with_options` required at least one option!

Reply:
I am 95% sure it does after reading through it and its call chain. In the next PR I will add some test cases to confirm the behavior and if necessary fix the implementation.